scripts/uberenv/packages/axom/package.py

```diff
@@ -96,15 +98,13 @@ class Axom(CMakePackage, CudaPackage):
     depends_on("mpi", when="+mpi")
 
     # Libraries
-    depends_on("conduit~shared+python", when="+python")
-    depends_on("conduit~shared~python", when="~python")
-    depends_on("conduit~shared+python+hdf5", when="+hdf5+python")
-    depends_on("conduit~shared+python~hdf5", when="~hdf5+python")
-    depends_on("conduit~shared~python+hdf5", when="+hdf5~python")
-    depends_on("conduit~shared~python~hdf5", when="~hdf5~python")
+    depends_on("conduit+python", when="+python")
+    depends_on("conduit~python", when="~python")
+    depends_on("conduit+hdf5", when="+hdf5")
+    depends_on("conduit~hdf5", when="~hdf5")
 
     # HDF5 needs to be the same as Conduit's
-    depends_on("hdf5@1.8.19:1.8.999~mpi~cxx~shared~fortran", when="+hdf5")
+    depends_on("hdf5@1.8.19:1.8.999~cxx~shared~fortran", when="+hdf5")
```

Comment:
I don't understand the fixed version here. Does spack not prevent you from using axom with conduit when you use out of sync hdf5? e.g.
```
spack install axom ^hdf5@1.12~cxx~shared~fortran
```
should install conduit with hdf5 1.12 (if conduit allows it) or fail (if conduit restricts the version).

Reply:
It has to do with how Spack locks in the concretization on a first come first serve nature.  Axom will get their first and default to the highest version, then conduit restricts that to the lower version and it fails.  If you don't all agree then you can't install both at the same time with one command like "spack install axom@develop"

Reply:
@jandrej for the record:
https://github.com/LLNL/serac/pull/48#discussion_r409999658

However, there might be a change of strategy coming... ;)

Reply:
Oops, that’s Axom...